static/js/protein/data_processing_utils.ts

```diff
@@ -592,3 +529,248 @@ export function getProteinDescription(data: GraphNodes): string {
+  };
+}
+
+/**
+ * Given response and key, map each response datum to value of key and return map
+ */
+export function getFromResponse<K extends keyof V1BioResponseDatum>(
+  resp: V1BioResponse,
+  key: K
+): V1BioResponseDatum[K][] {
+  if (!("data" in resp.data)) {
+    return [];
+  }
+  return resp.data.map((obj) => obj[key]);
+}
+
+/**
+ * Given an array of interaction DCIDs and a parallel array of corresponding scores,
+ * construct and return score record such that for each interaction A_B with corresponding DCID in interaction DCIDs,
+ * both A_B and B_A map to the score of A_B.
+ *
+ * Since we currently do not graphically distinguish between the A_B and B_A scores,
+ * we choose to have a symmetric score store.
+ *
+ * In the future perhaps we should make the graph directed to make relationships like "A phosphorylates B"
+ * more clear. I also think there should be assays for which the scores are not symmetric -
+ * ex. "knockout A and see what happens to the expression of B". These might not exist in the graph currently
+ * but could very well be imported in the future. Thus, we might have two or more edges between proteins
+ * for each type of interaction they participate in, with different scores for each.
+ */
+export function symmetricScoreRec(
+  interactionDcids: bioDcid[],
+  scores: number[]
+): Record<string, number> {
+  const scoreRec = {};
+  for (let i = 0; i < interactionDcids.length; i++) {
+    const [proteinA, proteinB] = proteinsFromInteractionDcid(
+      interactionDcids[i]
+    );
+    const scoreAB = _.get(
+      scoreRec,
+      `${proteinA}_${proteinB}`,
+      DEFAULT_INTERACTION_SCORE
+    );
+    const scoreBA = _.get(
+      scoreRec,
+      `${proteinB}_${proteinA}`,
+      DEFAULT_INTERACTION_SCORE
+    );
+    const maxScore = Math.max(scores[i], scoreAB, scoreBA);
+    scoreRec[`${proteinA}_${proteinB}`] = maxScore;
+    scoreRec[`${proteinB}_${proteinA}`] = maxScore;
+  }
+  return scoreRec;
+}
+
+/**
+ * Given score response, construct a record mapping interaction IDs of the form A_B to their confidence scores,
+ * satisfying the property that if A_B: A_B.score is in the record, then B_A: A_B.score is also.
+ */
+export function scoreDataFromResponse(
+  scoreResponse: V1BioResponse
+): Record<string, number> {
+  const scoreValues = getFromResponse(scoreResponse, "values");
+  const interactionDcids = getFromResponse(scoreResponse, "entity");
+  const scoreList = scoreValues
+    // map scoreValues to IntactMiScore if available or default interaction score
+    .map((bioData: V1BioDatum[]) => {
+      for (const bioDatum of bioData) {
+        if (_.get(bioDatum, "dcid", "").includes(INTERACTION_QUANTITY_DCID)) {
+          const score = quantityFromDcid(
+            bioDatum.dcid,
+            INTERACTION_QUANTITY_DCID
+          );
+          if (!isNaN(score)) {
+            return score;
+          }
+          console.warn(`Invalid quantity ID ${bioDatum.dcid} -- skipping`);
```

Comment:
I'm not sure putting the console.warn here makes the most sense either because you're still assigning a score here (DEFAULT_INTERACTION_SCORE), so for a user, it won't actually look like anything has been skipped? 

Maybe the console.warn can be added where this function is called and we can console.warn all cases where the score is DEFAULT_INTERACTION_SCORE & the message can be something like `"Unable to retrieve score for interaction <interaction dcid> -- default interaction score of <interaction score value> used"`

Reply:
Ohhh! Yes that's much better. I changed it to

```
scoreValues
    .map((bioData: V1BioDatum[], i) => {
      for (const bioDatum of bioData) {
        ...
      }
      console.warn(`Unable to retrieve score for interaction ${interactionDcids[i]} -- default score of ${DEFAULT_INTERACTION_SCORE} used`);
      return DEFAULT_INTERACTION_SCORE;
    })
```